vars/generateAndExecutePipeline.groovy

```diff
@@ -53,6 +57,17 @@ def _generateAndExecutePipeline() {
     }
 }
 
+def _assertValidBranchType() {
+
+    isReleaseOrPR = _isReleaseOrPr()
+    isFeature= _isFeature()
+
+    if (!isReleaseOrPR && !isFeature) {
```

Comment:
To better read this line of code, we can change it to: if (isReleaseOrPr || isFeature)

Reply:
Good idea.  CodeNarc would have :heart: this one.

Reply:
Hmmm.... I disagree. I think the proposed changes the functionality. We only want to throw the following error if the current branch is __not__ release, PR or feature branch. As I read the proposed change we would throw the error if the branch __is__ any of those things.

Am I missing something, or did you guys miss the `!` marks in the current expression?

`error("Only 'feature/', 'release/', 'PR-' and 'master' branches can be built.")`

Reply:
See [De Morgan's Laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)

A || B  is logically equivalent to !A && !B
A && B is logically equivalent to !A || !B

Reply:
well I totally read that wrong

Reply:
Hard to argue with De Morgan. Will change.

Reply:
I'm totally not reading right too, your suggestion misses the wrapping `!()`

```suggestion
    if (!(isReleaseOrPR || isFeature)) {
```

Reply:
It's been a while since I had to read formal language.

Reply:
Thanks guys